go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -279,6 +282,15 @@ func (bnc *BaseNetworkController) findPodWithIPAddresses(needleIPs []net.IP) (*k
 		if util.PodCompleted(p) || util.PodWantsHostNetwork(p) || !util.PodScheduled(p) {
 			continue
 		}
+
+		// If the network type is Layer 3, then IP allocation is per node, so we can filter pods by node name.
+		// If pod is not applicable to live migration, restrict the search to only pods on the filtered node
+		// Note we could check to see if pod has been migrated, but that search is even heavier than the
+		// alternative.
```

Comment:
Hmm not sure about this

This could all be subnet based right? We would just care about pods in the same subnet. And I wouldn't be sure if actually checking for subnet containment could be faster than comparing strings, the former could be less comparisons.

Anyway, I am fine with this as is an improvement in any case.

Reply:
from what I saw we would have to call  virtualMachineReady to know if the machine has migrated or not. It does a search based on pod selector, so maybe its not that bad. Either way I didnt really care that much because the optimizing for the live migration case is so small its not really worth it.

Reply:
I updated with a comment to mention the specific scenario im chasing after here, where the pod status IPs are not yet available so we want to avoid an unnecessary unmarshal if the pod is not in the same subnet as the one we are looking for.